### PR TITLE
[api] refactor call param parsing

### PR DESCRIPTION
**Pull request by envestcc**
# Description
as title

## Type of change
Please delete options that are not relevant.
- [] Bug fix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [x] Code refactor or improvement
- [] Breaking change (fix or feature that would cause a new or changed behavior of existing functionality)
- [] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [] make test
- [] fullsync
- [] Other test (please specify)

**Test Configuration**:
- Firmware version:
- Hardware:
- Toolchain:
- SDK:

# Checklist:
- [] My code follows the style guidelines of this project
- [] I have performed a self-review of my code
- [] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [] My changes generate no new warnings
- [] I have added tests that prove my fix is effective or that my feature works
- [] New and existing unit tests pass locally with my changes
- [] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by CoderZhi — APPROVED** [review]

File: api/web3server.go [hunk 1]
```diff
@@ -435,11 +436,11 @@ func (svr *web3Handler) call(in *gjson.Result) (interface{}, error) {
 }
 
 func (svr *web3Handler) estimateGas(in *gjson.Result) (interface{}, error) {
-	from, to, gasLimit, gasPrice, value, data, err := parseCallObject(in)
+	callMsg, err := parseCallObject(in)
 	if err != nil {
 		return nil, err
 	}
-
+	from, to, gasLimit, gasPrice, value, data := callMsg.From, callMsg.To, callMsg.Gas, callMsg.GasPrice, callMsg.Value, callMsg.Data
```
Comment (dustinxie) [inline api/web3server.go:443]: from, to := callMsg.From, callMsg.To
the other 4 can directly use `callMsg.xxx` at L457 below

File: api/web3server.go [hunk 2]
```diff
@@ -1085,12 +1086,14 @@ func (svr *web3Handler) traceCall(ctx context.Context, in *gjson.Result) (interf
...
 		callerAddr   address.Address
+		callMsg      *callMsg
 	)
 	blkNumOrHashObj, options := in.Get("params.1"), in.Get("params.2")
-	callerAddr, contractAddr, gasLimit, _, value, callData, err = parseCallObject(in)
+	callMsg, err = parseCallObject(in)
 	if err != nil {
 		return nil, err
 	}
+	callerAddr, contractAddr, gasLimit, value, callData = callMsg.From, callMsg.To, callMsg.Gas, callMsg.Value, callMsg.Data
```
Comment (dustinxie) [inline api/web3server.go:1096]: same here, can directly use `callMsg.xxx` at L1136

File: api/web3server.go (hunk 1 above)
Comment (envestcc) [reply api/web3server.go]: fixed

File: api/web3server.go (hunk 2 above)
Comment (envestcc) [reply api/web3server.go]: fixed

**Review by huangzhiran — APPROVED** [review]

**Pull request merged**